### ISSUE TITLE
Weight decay bug in textual_inversion.py

### ISSUE THREAD

**Issue by scoutsaachi** [huggingface/diffusers#1009]
### Describe the bug

I think there's a bug in the [textual_inversion.py ](https://github.com/huggingface/diffusers/blob/main/examples/textual_inversion/textual_inversion.py)script. Specifically, the word embeddings of all the tokens except the newly added token are supposed to be frozen. The way the script does this is by zeroing out the gradients for everything except the last token.

However, this still subjects the pre-trained word embeddings to weight decay. This means that these "frozen" embeddings are still changing, and with enough training will decay to 0. 

### Reproduction

I just printed out the embeddings at the beginning of training and after 3000 steps, and they were slightly different. 
Before:
tensor([[-0.0012,  0.0368,  0.0221,  ...,  0.0158,  0.0046, -0.0219],
        [ 0.0152,  0.0262, -0.0132,  ..., -0.0037,  0.0002,  0.0121],
        [-0.0154, -0.0131,  0.0065,  ..., -0.0206, -0.0139, -0.0025],
        ...,
        [ 0.0011,  0.0032,  0.0003,  ..., -0.0018,  0.0003,  0.0019],
        [ 0.0012,  0.0077, -0.0011,  ..., -0.0015,  0.0009,  0.0052],
        [ 0.0434,  0.0019,  0.0327,  ...,  0.0250, -0.0058,  0.1477]],
       device='cuda:0', dtype=torch.float16, requires_grad=True)

After:
tensor([[-0.0011,  0.0347,  0.0208,  ...,  0.0149,  0.0043, -0.0207],
        [ 0.0143,  0.0247, -0.0124,  ..., -0.0035,  0.0002,  0.0114],
        [-0.0145, -0.0124,  0.0061,  ..., -0.0194, -0.0131, -0.0024],
        ...,
        [ 0.0010,  0.0030,  0.0003,  ..., -0.0017,  0.0003,  0.0018],
        [ 0.0011,  0.0073, -0.0010,  ..., -0.0014,  0.0008,  0.0049],
        [ 0.0434,  0.0019,  0.0327,  ...,  0.0250, -0.0058,  0.1477]],
       device='cuda:0', dtype=torch.float16, requires_grad=True)

As you can see, the values are decaying slightly. 

### System Info

- `diffusers` version: 0.6.0
- Platform: Linux-4.15.0-193-generic-x86_64-with-glibc2.27
- Python version: 3.9.12
- PyTorch version (GPU?): 1.11.0 (True)
- Huggingface_hub version: 0.10.1
- Transformers version: 4.23.1
- Using GPU in script?:  YES
- Using distributed or parallel set-up in script?: NO

**Comment by patrickvonplaten** [comment]
@patil-suraj could take a look here?

**Comment by shunk031** [comment]
There is a similar issue: https://github.com/huggingface/diffusers/issues/507 👀

**Comment by patrickvonplaten** [comment]
Repinging @patil-suraj

**Comment by patrickvonplaten** [comment]
This should be solved now in https://github.com/huggingface/diffusers/pull/1665
